app/src/main/java/com/example/plantsapp/presentation/ui/notification/TaskNotificationManager.kt

```diff
@@ -78,11 +85,29 @@ class TaskNotificationManager @Inject constructor(
         )
     }
 
+    private fun createCompletePendingIntent(task: Task, notificationId: Int): PendingIntent {
+        val completeIntent = Intent(context, NotificationBroadcastReceiver::class.java)
+            .apply {
+                this.notificationId = notificationId
+                taskId = task.id
+                taskFrequency = task.frequency
+                taskKey = TaskKeys.from(task).key
+            }
+        return PendingIntent.getBroadcast(
+            context,
+            completeRequestCode++,
+            completeIntent,
+            PendingIntent.FLAG_IMMUTABLE or PendingIntent.FLAG_UPDATE_CURRENT
+        )
+    }
+
     private fun prepareTaskNotification(
         plantName: String,
         plantPicture: Bitmap,
         task: Task
-    ): Notification {
+    ): Pair<Notification, Int> {
+        val id = notificationId++
```

Comment:
Same here - lets try to move this thing into parameters, to give id from outer scope